From 97787d7894e363aed2b45d78bc21ad0d09a90c12 Mon Sep 17 00:00:00 2001
From: Priyanka Bose <priyanka.bose@intel.com>
Date: Sun, 31 Dec 2017 04:16:53 +0530
Subject: [PATCH 1/2] Adding dependancies for Capstone Disassembler in Vendor
 Art

Change-Id: I5a8c18c8833941020ae24285ab516e28b06fcd1c
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-55911
Category: device enablement
Domain: AOSP.ART-Other
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Priyanka Bose <priyanka.bose@intel.com>
---
 android/config.go   | 4 ++++
 android/variable.go | 9 +++++++++
 2 files changed, 13 insertions(+)

diff --git a/android/config.go b/android/config.go
index 5dcc59b..dc46381 100644
--- a/android/config.go
+++ b/android/config.go
@@ -548,3 +548,7 @@ func (c *config) IntegerOverflowDisabledForPath(path string) bool {
 	}
 	return prefixInList(path, *c.ProductVariables.IntegerOverflowExcludePaths)
 }
+
+func (c *config) AutoFastJni() bool {
+        return Bool(c.ProductVariables.AutoFastJni)
+}
diff --git a/android/variable.go b/android/variable.go
index 77bdda2..e268a20 100644
--- a/android/variable.go
+++ b/android/variable.go
@@ -94,6 +94,13 @@ type variableProperties struct {
 		Pdk struct {
 			Enabled *bool
 		}
+
+		AutoFastJni struct {
+                        Srcs          []string
+                        Include_dirs  []string
+                        Static_libs   []string
+                        Shared_libs   []string
+                }
 	} `android:"arch_variant"`
 }
 
@@ -167,6 +174,8 @@ type productVariables struct {
 	Override_rs_driver *string `json:",omitempty"`
 
 	DeviceKernelHeaders []string `json:",omitempty"`
+
+	AutoFastJni *bool `json:",omitempty"`
 }
 
 func boolPtr(v bool) *bool {
-- 
1.9.1

